File: BW/10월3주차/멀리뛰기.js
// 출력 예시와 문제 스타일이 이 전 피보나치와 비슷한 점에서 착안

function solution(n) {
    var fibo = [];
    fibo[0] = 1, fibo[1] = 1;
    
    for(i=2; i<=n; i++){
        fibo[i] = (fibo[i-1] + fibo[i-2])%1234567;
    }

    return fibo[n];
}

File: JY/10월 3주차/멀리뛰기.py
"""
N = 1 -> 1칸 
N = 2 -> 2칸 
N = 3 -> 2칸까지 온 경우의 수 1칸 까지 온 경우의 수
N = 4 -> 3칸까지 온 경우의 수 + 2칸까지 온 경우의 수
"""
MAX_VALUE = 2001
MOD = 1234567
def solution(n):
    answer = 0
    dp = [0 for _ in range(MAX_VALUE)]
    dp[1] = 1
    dp[2] = 2
    
    for i in range(3, MAX_VALUE):
        dp[i] = (dp[i-1] + dp[i-2]) % MOD
        
    
    answer = dp[n]%MOD
    return answer